src/libostree/ostree-repo-commit.c

```diff
@@ -1057,6 +1093,29 @@ ostree_repo_prepare_transaction (OstreeRepo     *self,
   memset (&self->txn_stats, 0, sizeof (OstreeRepoTransactionStats));
 
   self->in_transaction = TRUE;
+  if (self->min_free_space_percent > 0)
+    {
+      struct statvfs stvfsbuf;
+      if (TEMP_FAILURE_RETRY (fstatvfs (self->repo_dir_fd, &stvfsbuf)) < 0)
+        return glnx_throw_errno_prefix (error, "fstatvfs");
+      g_mutex_lock (&self->txn_stats_lock);
+      self->txn_blocksize = stvfsbuf.f_bsize;
+      /* Convert fragment to blocks to compute the total */
+      guint64 total_blocks = (stvfsbuf.f_frsize * stvfsbuf.f_blocks) / stvfsbuf.f_bsize;
```

Comment:
Doesn't this multiplication yield the total size of the filesystem in bytes already? `statvfs(2)` says:

```
fsblkcnt_t     f_blocks;   /* Size of fs in f_frsize units */
```

Writing a quick program to check this:

```
$ cat tmp.c                    
#include <sys/statvfs.h>                         
#include <stdio.h>      

int main(void)          
{                       
    struct statvfs stat;                         
    if (statvfs("/", &stat) < 0)                 
        return 1;       

    unsigned long long total_bytes = stat.f_blocks * stat.f_frsize;                                
    printf("total size in bytes: %llu\n", total_bytes);                                            
    printf("total size in GB: %llu\n", total_bytes >> 30);                                         

    return 0;           
}                       
$ ./tmp                                                                          
total size in bytes: 95133081600                 
total size in GB: 88    
$ df -hT /                                                                       
Filesystem              Type  Size  Used Avail Use% Mounted on                                     
/dev/mapper/fedora-root ext4   89G   63G   22G  75% /          
```

So then, we need to divide this by `f_bsize` to be made comparable to `bfree`, right?